### PR TITLE
Display a message in a table row if there is no data to display

### PR DESCRIPTION
**Pull request by joannecheng**
## What does this PR do? How does it affect users?

This PR displays a message when empty tables are rendered in keen-dataviz. This way a user doesn't need to look at a blank slate every time they load a table with no data.

## How should this be tested?

Look at the "Empty Table" example in `test/demo/index.html`. Make sure a message is displayed for an empty dataset.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [ ] Does the code compile without warnings (check shell, console)?
- [ ] Do all tests pass?
- [ ] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [ ] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [ ] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [ ] If the feature saves data to a database, can you confirm the data is indeed created in the database?

## Related tickets?

https://metakeen.atlassian.net/secure/RapidBoard.jspa?rapidView=28&view=detail&selectedIssue=EXP-469

### Before
screen shot 2016-10-31 at 1 46 19 pm

### After
screen shot 2016-12-20 at 1 45 25 pm

### CODE REVIEW

**Comment by dustinlarimer** [comment]
Other viz types use the `.message()` method to communicate an empty response: https://github.com/keen/keen-dataviz.js/blob/master/lib/libraries/index.js#L129-L132. What do you think about keeping with that as a pattern?

```javascript
if (this.data()[0].length === 1 || this.data().length === 1) {
  this.message('No data to display');
  return;
}
```

**Comment by joannecheng** [comment]
yes! I totally missed that. Thanks for picking that up

**Comment by joannecheng** [comment]
@dustinlarimer I really like the message display, but unfortunately we lose the table title, which was one of the complaints of the KP customer. Is there a way to display that message without losing the title?

**Comment by dustinlarimer** [comment]
Thats a good point, should we just update that message type to include the
title and footnotes?

**Comment by joannecheng** [comment]
Sounds good, I can work on that! 🎉

**Comment by animatecitizen** [comment]
Wiki Traders Data

**Comment by animatecitizen** [comment]
Wki Traders My data

**Comment by joannecheng** [comment]
Hi @dustinlarimer, could I get a re-review? Using the `message` method to show empty messages now.

File: lib/libraries/default/message.js [hunk 1]
```diff
@@ -2,20 +2,31 @@ var escapeHtml = require('../../utils/escape-html');
...
+    var inner = document.createElement('div');
+    var msg = document.createElement('span');
+    var height = this.height() || 140;
+
+    var titleContainer = document.createElement('div');
+    var notesContainer = document.createElement('div');
 
     outer.className = this.theme();
     inner.className = this.theme() + '-message';
     inner.style.height = height + 'px';
```
Comment (dustinlarimer) [inline lib/libraries/default/message.js]: We need to factor in `this.el().offsetHeight` here so that the title and notes elements' height (and borders, margins, padding, etc) will be accounted for in the final height. Here's how it's done for C3.js artifacts: https://github.com/keen/keen-dataviz.js/blob/d02dfb6fcbcf125bdfbde186409990e0f1df93c1/lib/libraries/index.js#L103

Comment (dustinlarimer) [inline lib/libraries/default/message.js]: Here's another approach, from `table.js`: https://github.com/keen/keen-dataviz.js/blob/d02dfb6fcbcf125bdfbde186409990e0f1df93c1/lib/libraries/default/table.js#L29

**Comment by joannecheng** [comment]
@dustinlarimer Thanks for catching that! I had some trouble with using `this.el().offsetHeight` (can explain more in slack), so I calculated the correct height for `inner.style.height` by subtracting `titleContainer.offsetHeight` and `notesContainer.offsetHeight` from the given `height` variable. It still works as expected though:

screen shot 2016-12-20 at 3 58 18 pm

**Comment by dustinlarimer** [comment]
👍 awesome, that works!
